### PR TITLE
Update class-wc-gateway-cardknox.php

### PR DESCRIPTION
**Pull request by mstein-ck**

### CODE REVIEW

**Review by ymwymw — COMMENTED** [review]
Waiting on `isset` change

**Comment by mstein-ck** [comment]
@pritesh-lr can you review?

**Comment by mstein-ck** [comment]
I'm not sure what the correct Exception type should be.

**Review by ymwymw — APPROVED** [review]

File: includes/class-wc-gateway-cardknox.php [hunk 1]
```diff
@@ -416,8 +416,24 @@ public function get_payment_data($post_data){
...
+  }
+
+  public function validate_payment_data($post_data){
+    if ($this->is_unset_or_empty($post_data['xToken'])) {
+      if ($this->is_unset_or_empty($post_data['xCardNum'])) {
+        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Invalid card number", 400);
+      }
+      if ($this->is_unset_or_empty($post_data['xCVV'])) {
+        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Invalid cvv", 400);
+      }
```
Comment (ymwymw) [inline includes/class-wc-gateway-cardknox.php:430]:
```suggestion
        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Required: card number", 400);
      }
      if ($this->is_unset_or_empty($post_data['xCVV'])) {
        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Required: cvv", 400);
      }
```

**Review by ymwymw — COMMENTED** [review]
See comment

File: includes/class-wc-gateway-cardknox.php (hunk 1 above)
Comment (pritesh-lr) [reply includes/class-wc-gateway-cardknox.php:430]: Hello @ymwymw ,

yes, Exception codes are valid. 
I tried to pull latest master on my server, but it still shows old code.
I think all latest code is not merged to master yet.

Can you please merge on master and confirm ?
so I can download that latest source and also do testing with test site.

**Review by ymwymw — APPROVED** [review]

File: includes/class-wc-gateway-cardknox.php [hunk 2]
```diff
@@ -416,8 +416,24 @@ public function get_payment_data($post_data){
...
+      if ($this->is_unset_or_empty($post_data['xCVV'])) {
+        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Required: cvv", 400);
+      }
     }
+  }
+
+  private function is_unset_or_empty($s){
+    return !isset($s) || $s === '';
+  }
```
Comment (ymwymw) [inline includes/class-wc-gateway-cardknox.php:436]:
```suggestion
    private function is_unset_or_empty($s){
      return !isset($s) || $s === '';
    }
```

**Review by ymwymw — APPROVED** [review]

**Pull request merged**